curve25519-dalek/src/backend/serial/u64/field_verus.rs

```diff
@@ -395,29 +324,95 @@ impl FieldElement51 {
         // First, reduce the limbs to ensure h < 2*p.
         let mut limbs = FieldElement51::reduce(self.limbs).limbs;
 
+        // Compute q with overflow proofs
+        // After reduce, limbs[i] < 2^52, so limbs[0] + 19 < 2^52 + 19 < 2^64
+        proof {
+            assert((1u64 << 52) + 19 <= u64::MAX) by (compute);
+        }
```

Comment:
Not a fan of interweaving proof blocks and exec code. TODO: investigate if a proof can be isolated in a top proof block or externally in a lemma.

Reply:
Postponed, see #191